### ISSUE TITLE
hlsMuxer crashes when using h264_v4l2m2m encoder on Raspberry PI 4

### ISSUE THREAD

**Issue by georgy-komarov** [aler9/rtsp-simple-server#1155]
## Which version are you using?

v0.20.0

## Which operating system are you using?

- [ ] Linux amd64 standard
- [ ] Linux amd64 Docker
- [x] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

I'm trying to stream USB Webcam video as LL-HLS and using hardware h264 encoding on Raspberry Pi 4 (Raspbian arm64) with `h264_v4l2m2m`. However rtsp-simple-server crashes in 1-2 seconds after ffmpeg start.

However I'm not sure if it is ffmpeg issue or rtsp-simple-server issue because everything looks fine if I run the following commands
```
ffmpeg -f v4l2 -video_size 640x480 -framerate 24 -i /dev/video0 -c:v h264_v4l2m2m -an crash.mp4
ffmpeg -re -stream_loop -1 -i crash.mp4 -c copy -f rtsp rtsp://localhost:8554/cam
```

## Describe how to replicate the issue

1. Start the server
2. Publish with `ffmpeg -f v4l2 -video_size 640x480 -framerate 24 -i /dev/video0 -c:v h264_v4l2m2m -an -f rtsp rtsp://localhost:8554/cam`

## Did you attach the server logs?

yes
[crash.log](https://github.com/aler9/rtsp-simple-server/files/9576412/crash.log)

## Did you attach a network dump?

no

**Comment by aler9** [comment]
Hello, this issue is caused by two bugs:

1) a FFmpeg-side bug: when using `h264_v4l2m2m`, the SPS and PPS are not sent to the server. This has been fixed in FFmpeg 5 as reported by @roger- . FFmpeg 5 can be downloaded [here](https://johnvansickle.com/ffmpeg/)

2) a server-side bug: when SPS and PPS are not present, two empty NALUs are prepended to any IDR NALU, resulting in the crash

**Comment by aler9** [comment]
Fixed in v0.21.1
